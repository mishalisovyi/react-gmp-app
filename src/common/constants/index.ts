export const Icon = {
  VERTICAL_ELLIPSIS: String.fromCharCode(8942),
  MULTIPLICATION_X: String.fromCharCode(10005),
  TELEPHONE_RECORDER: String.fromCharCode(8981),
};

export const API_URL = 'http://localhost:4000';

export const HTTP_RESPONSE_STATUS_NO_CONTENT = 204;
