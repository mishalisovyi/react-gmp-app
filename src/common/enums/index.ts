export { ButtonType } from './button-type.enum';
export { HttpMethod } from './http-method.enum';
