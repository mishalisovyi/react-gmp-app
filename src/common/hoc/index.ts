export { withPortalRendering } from './withPortalRendering/withPortalRendering';
export { ModalWindow } from './ModalWindow/ModalWindow';
