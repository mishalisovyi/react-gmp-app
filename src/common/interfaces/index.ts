export { EnumerableComponentProps } from './EnumerableComponentProps';
export { ModalWindowProps, ModalWindowActionButtonProps } from './ModalWindowProps';
export { SelectListData, SelectListValue } from './SelectListData';
