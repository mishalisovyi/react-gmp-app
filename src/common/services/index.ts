export {
  sendGetRequest,
  sendPostRequest,
  sendPutRequest,
  sendDeleteRequest,
} from './http.service';
