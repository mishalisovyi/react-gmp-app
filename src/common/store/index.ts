export * from './actions';
export * from './models';
export { commonReducer } from './reducer';
