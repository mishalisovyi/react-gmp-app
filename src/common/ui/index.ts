export { Button } from './Button/Button';
export { Divider } from './Divider/Divider';
export { DropdownMenu } from './DropdownMenu/DropdownMenu';
export { FilterSelectList } from './FilterSelectList/FilterSelectList';
export { LabelPrimary } from './LabelPrimary/LabelPrimary';
export { LabelSecondary } from './LabelSecondary/LabelSecondary';
export { Logo } from './Logo/Logo';
export { PopupConfirmation } from './PopupConfirmation/PopupConfirmation';
export { PopupDialog } from './PopupDialog/PopupDialog';
export { PopupInfo } from './PopupInfo/PopupInfo';
export { SearchPanel } from './SearchPanel/SearchPanel';
export { Spinner } from './Spinner/Spinner';
export { Tab } from './TabBar/Tab/Tab';
export { TabBar } from './TabBar/TabBar';
export { TextField } from './TextField/TextField';
