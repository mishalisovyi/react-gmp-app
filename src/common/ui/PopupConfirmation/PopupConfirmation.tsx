import React, { MouseEventHandler, useMemo } from 'react';

import { ModalWindow } from 'common/hoc';
import { ModalWindowActionButtonProps } from 'common/interfaces';

interface PopupConfirmationProps {
  title: string;
  message: string;
  onConfirm: MouseEventHandler<HTMLButtonElement>;
  onClose: MouseEventHandler<HTMLButtonElement>;
}

export function PopupConfirmation({
  title, message, onConfirm, onClose,
}: PopupConfirmationProps) {
  const primaryButtonAttributes: ModalWindowActionButtonProps = useMemo(() => ({
    show: true,
    text: 'Confirm',
    onClick: onConfirm,
  }), [onConfirm]);

  return (
    <ModalWindow title={title} primaryButton={primaryButtonAttributes} onClose={onClose}>
      <p className="typography-body">{message}</p>
    </ModalWindow>
  );
}
