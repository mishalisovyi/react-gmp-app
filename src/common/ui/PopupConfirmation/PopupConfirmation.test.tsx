import React, { ReactNode, ReactPortal } from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Icon } from 'common/constants';

import { PopupConfirmation } from './PopupConfirmation';

const handlePopupConfirming = jest.fn();
const handlePopupClosing = jest.fn();

beforeAll(() => {
  ReactDOM.createPortal = (node: ReactNode): ReactPortal => node as ReactPortal;
});

it('should render confirmation popup', () => {
  const tree = renderer.create(<PopupConfirmation title="test title" message="test message" onConfirm={handlePopupConfirming} onClose={handlePopupClosing} />);

  expect(tree.toJSON()).toMatchSnapshot();
});

it('should handle confirmation popup confirming', async () => {
  render(<PopupConfirmation title="test title" message="test message" onConfirm={handlePopupConfirming} onClose={handlePopupClosing} />);

  await userEvent.click(screen.getByText('Confirm'));

  expect(handlePopupConfirming).toHaveBeenCalled();
});

it('should handle confirmation popup closing', async () => {
  render(<PopupConfirmation title="test title" message="test message" onConfirm={handlePopupConfirming} onClose={handlePopupClosing} />);

  await userEvent.click(screen.getByText(Icon.MULTIPLICATION_X));

  expect(handlePopupClosing).toHaveBeenCalled();
});
