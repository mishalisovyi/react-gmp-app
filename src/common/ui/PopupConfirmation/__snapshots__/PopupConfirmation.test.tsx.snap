// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render confirmation popup 1`] = `
<div>
  <div />
  <div>
    <div>
      <button
        className="button button--transparent"
        onClick={[MockFunction]}
        type="button"
      >
        âœ•
      </button>
    </div>
    <header>
      test title
    </header>
    <div>
      <p
        className="typography-body"
      >
        test message
      </p>
    </div>
    <div>
      <button
        className="button button--primary"
        onClick={[MockFunction]}
        type="button"
      >
        Confirm
      </button>
    </div>
  </div>
</div>
`;
